### DISCUSSION TITLE
How to maintain style components while using a Custom Value Editor

### DISCUSSION THREAD

**Discussion by rbishop3757** [react-querybuilder/react-querybuilder#518] [Q&A] [answered]
Hello! I am utilizing the react-query-builder with MUI styling and attempting to create a custom value editor for only a specific type of field. The CustomValueEditor returns either a component I have made if the specific type of field is used, or a generic `ValueEditor` with all props passed into it. Whenever I pass in the CustomValueEditor to the `QueryBuilder`, the MUI styling is lost. Is there a way to maintain that styling for the `ValueEditor` through the parent container `QueryBuilderMaterial`? Thanks!

**Comment by jakeboone02** [comment, marked as answer]
Hi @rbishop3757! You need to

```
return <MaterialValueEditor {...props} />
```

instead of `ValueEditor`. You can

```
import { MaterialValueEditor } from "@react-querybuilder/material"
```

**Reply by rbishop3757** [reply]
Thank you so much! This is exactly what I was looking for.
